File: 2099-number-of-strings-that-appear-as-substrings-in-word/2099-number-of-strings-that-appear-as-substrings-in-word.cpp
class Solution {
public:
    int numOfStrings(vector<string>& patterns, string word)
    {

        int flag = 1;
        int ans = 0;

     for(int i = 0; i < patterns.size(); i++)
     {
         string str = patterns[i];

         if(str.length() <= word.length())
         {
             int k = 0;
             while(k < (word.length() - str.length() + 1) )
             {
                 if(str[0] == word[k])
                 {
                     flag = 2;
                     int first = 1;
                     int second = k;
                     second++;

                    while(first < str.length())
                    {
                        if(str[first] == word[second])
                        {
                            flag = 2;
                            first++;
                            second++;
                        }
                        else
                        {
                            flag = 1;
                            break;
                        }
                    } 
                    if(flag == 2)
                    {
                        ans++;
                        break;
                    }

                 }

                 k++;
             }

         }
         
     }

     return ans;
    }
};

File: 2099-number-of-strings-that-appear-as-substrings-in-word/2099-number-of-strings-that-appear-as-substrings-in-word.java
class Solution {
    public int numOfStrings(String[] patterns, String word) {
        
        int cnt = 0;
        // using stl
        for(String str : patterns)
        {
            // if(word.contains(str))
            if(word.indexOf(str) >= 0)
            {
                cnt++;
            }
        }

        // for(String str : patterns)
        // {
        //     if(str.length() > word.length())
        //     {
        //         continue;
        //     }

        //     boolean flag = false;
        //     int k = 0;
            
        //     for(int i = 0; i < word.length(); i++)
        //     {
        //         if(str.charAt(k) == word.charAt(i))
        //         {
        //             flag = true;
        //             k++;
        //             i++;
        //             while(k < str.length())
        //             {
                        
        //                 if( str.charAt(k) == word.charAt(i))
        //                 {
        //                 i++;
        //                 k++;
        //                 }
        //                 else if(str.charAt(k) != word.charAt(i))
        //                 {
        //                     flag = false;
        //                     break;
        //                 }
        //             }
        //             break;
        //         }
            
        //     }

        //     if(flag)
        //     {
        //         System.out.println(str);
        //         cnt++;
        //     }
        // }

        return cnt;
    }
}